### ISSUE TITLE
[Global] Controls are missing roles

### ISSUE THREAD

**Issue by svinkle** [CovidShield/mobile#4]
## Issue summary

Throughout the app, each clickable control is missing a role description. A role helps to describe what the control is, how to interact with it, and the expected outcome. Navigating the app without role descriptions may lead to a confusing user experience.

#### TSX

```tsx
// Button.tsx

if (Platform.OS === 'android') {
  return (
    <Ripple
      …
      onPress={onPressHandler}
    >
      {content}
    </Ripple>
  );
}
return (
  <TouchableOpacity
    onPress={onPressHandler}
    …
  >
    {content}
  </TouchableOpacity>
);
```

## Steps to reproduce

1. Install the CovidShield `apk` on an Android device
2. Load the app
3. Enable [TalkBack screen reader](https://support.google.com/accessibility/android/answer/6007100?hl=en)
4. Swipe left/right to navigate between content items

### Expected

- Controls to be announce with a role description.

### Actual

- Role descriptions are missing.

## Recommendation

1. For each component with an `onPress` prop, include the [`accessibilityRole`](https://reactnative.dev/docs/accessibility#accessibilityrole-android-ios) prop.
2. Set an appropriate value based on the purpose of the control:
   - If its a link, set as `link` <small>(Does the control request a resource, shift focus, or link out to an external web propert?)</small>
   - If its a button, set as `button` <small>(Does the control show/hide content, change onscreen data?)</small>

#### TSX

```tsx
// Button.tsx

if (Platform.OS === 'android') {
  return (
    <Ripple
      …
      onPress={onPressHandler}
      accessibilityRole={role}
    >
      {content}
    </Ripple>
  );
}
return (
  <TouchableOpacity
    onPress={onPressHandler}
    …
    accessibilityRole={role}
  >
    {content}
  </TouchableOpacity>
);
```

## Specifications

- **Component:** CovidShield [Global]
- **WCAG Principle:** Robust
- **WCAG SC:** 4.1.2 Name, Role, Value
- **Severity:** High
- **Effort:** Medium

**Comment by jordanAtShopify** [comment]
This has been fixed

**Comment by svinkle** [comment]
@jordanAtShopify This has not been fixed. I'm testing on `master` and many controls are still missing their `accessibilityRole` prop.

<details>
  <summary>Annotated screenshots</summary>

![Screen Shot 2020-06-08 at 10 03 19 AM](https://user-images.githubusercontent.com/1392632/84043133-ed8d1200-a973-11ea-8ba9-17531af75b3d.png)
![Screen Shot 2020-06-08 at 10 03 26 AM](https://user-images.githubusercontent.com/1392632/84043137-ee25a880-a973-11ea-8285-61f780d0d5cc.png)
![Screen Shot 2020-06-08 at 10 03 42 AM](https://user-images.githubusercontent.com/1392632/84043138-ee25a880-a973-11ea-9c72-612250c745d7.png)
![Screen Shot 2020-06-08 at 10 03 50 AM](https://user-images.githubusercontent.com/1392632/84043139-ee25a880-a973-11ea-9a40-b885f5055ddb.png)
![Screen Shot 2020-06-08 at 10 03 55 AM](https://user-images.githubusercontent.com/1392632/84043140-eebe3f00-a973-11ea-87f2-11a85a2796dc.png)
![Screen Shot 2020-06-08 at 10 04 04 AM](https://user-images.githubusercontent.com/1392632/84043141-eebe3f00-a973-11ea-8c3f-1329ab4c34d5.png)
![Screen Shot 2020-06-08 at 10 04 10 AM](https://user-images.githubusercontent.com/1392632/84043142-eebe3f00-a973-11ea-84fb-cbefd43ff490.png)
![Screen Shot 2020-06-08 at 10 04 43 AM](https://user-images.githubusercontent.com/1392632/84043144-eebe3f00-a973-11ea-84c1-75fe016bcc37.png)
![Screen Shot 2020-06-08 at 10 05 51 AM](https://user-images.githubusercontent.com/1392632/84043145-ef56d580-a973-11ea-9cd4-ddc5e7cc8ba6.png)
![Screen Shot 2020-06-08 at 10 06 04 AM](https://user-images.githubusercontent.com/1392632/84043146-ef56d580-a973-11ea-83c6-965ff70cf9a8.png)
![Screen Shot 2020-06-08 at 10 06 26 AM](https://user-images.githubusercontent.com/1392632/84043147-ef56d580-a973-11ea-9ac1-a729c6566b73.png)
![Screen Shot 2020-06-08 at 10 06 31 AM](https://user-images.githubusercontent.com/1392632/84043148-efef6c00-a973-11ea-8141-1d7cfaf334fd.png)
![Screen Shot 2020-06-08 at 10 06 37 AM](https://user-images.githubusercontent.com/1392632/84043150-efef6c00-a973-11ea-8ad4-004e80e0258f.png)
![Screen Shot 2020-06-08 at 10 09 23 AM](https://user-images.githubusercontent.com/1392632/84043152-f0880280-a973-11ea-8528-b7f3a0b97f00.png)

</details>

Some of these _have_ been fix, but I wanted to note all that I could see in the Figma file.

**Comment by jordanAtShopify** [comment]
@svinkle Sorry about that, the Button component now always has the proper role so some of these mist be their own Touchable. I'll reopen this a put a fix up.
